agentic-tutorial/src/main/java/_3_loop_workflow/_3a_Loop_Agent_Example.java

```diff
@@ -0,0 +1,88 @@
+package _3_loop_workflow;
+
+import dev.langchain4j.agentic.AgenticServices;
+import dev.langchain4j.agentic.UntypedAgent;
+import dev.langchain4j.model.chat.ChatModel;
+import domain.CvReview;
+import util.ChatModelProvider;
+import util.StringLoader;
+import util.log.CustomLogging;
+import util.log.LogLevels;
+
+import java.io.IOException;
+import java.util.Map;
+
+public class _3a_Loop_Agent_Example {
+
+    static {
+        CustomLogging.setLevel(LogLevels.PRETTY, 300);  // control how much you see from the model calls
+    }
+
+    /**
+     * This example demonstrates how to implement a CvReviewer agent that we can add to a loop
+     * with our CvTailor agent. We will implement two agents:
+     * - ScoredCvTailor (takes in a CV and tailors it to a CvReview (feedback/instruction + score))
+     * - CvReviewer (takes in the tailored CV and job description, and returns a CvReview object (feedback + score)
+     * Additionally, the loop ends when the score is above a certain threshold (e.g. 0.7) (exit condition)
+     */
+
+    // 1. Define the model that will power the agents
+    private static final ChatModel CHAT_MODEL = ChatModelProvider.createChatModel();
+
+    public static void main(String[] args) throws IOException {
+
+        // 2. Define the two sub-agents in this package:
+        //      - CvTailor.java
+        //      - CvReviewer.java
+
+        // 3. Create all agents using AgenticServices
+        ScoredCvTailor scoredCvTailor = AgenticServices.agentBuilder(ScoredCvTailor.class)
+                .chatModel(CHAT_MODEL)
+                .outputName("cv") // this will be updated in every iteration, continuously improving the CV
+                .build();
+        CvReviewer cvReviewer = AgenticServices.agentBuilder(CvReviewer.class)
+                .chatModel(CHAT_MODEL)
+                .outputName("cvReview") // this gets updated in every iteration with new feedback for the next tailoring
+                .build();
+
+        // 4. Build the sequence
+        UntypedAgent reviewedCvGenerator = AgenticServices // use UntypedAgent unless you define the resulting composed agent, see _2_Sequential_Agent_Example
+                .loopBuilder().subAgents(scoredCvTailor, cvReviewer) // this can be as many as you want, order matters
```

Comment:
Not sure if this would be acceptable in this example, but I believe that if you invert the order of these 2 agents you could avoid passing a fake `CvReview`, because in this way the `cvReviewer` will create the first one for the `scoredCvTailor`.

Related to this, at the moment the whole application fails with a `RuntimeException` if an agent requires an input that it is not present in the `AgenticScope`. I'm not sure if this behavior fits all use cases or in some situation you may want just to ignore the missing value and invoke the agent regardless. Any feedback on this?